cms/magento/magento2.php

```diff
@@ -0,0 +1,373 @@
+<?php
```

Comment:
This file contains all config & tasks for Magento 2, which is certainly a workable solution. I would like to request that this file gets renamed to `2.php` (and documentation updated) to help keep the API minimally consistent with the other supported platforms.

The config files for the other platforms only include base configuration + the definition of the `deploy` task. I find it _useful_ -- but not a requirement -- to have _one_ task defined per file in a namespaced directory similar to namespaced PHP classes. It helps separate logic into smaller chunks of code that are easier to analyze. This is merely a code style preference, though. I am fine with merging this code as-is. For the sake of consistency we should come to a consensus whether tasks should be defined in a single file or multiple namespaced files & implement the approach across the board.

Please let me know your thoughts on the matter. Thanks!

Reply:
- This topic is a part of a longer conversation we had with @colinodell , in which we weighted the pros and cons of splitting code into multiple files.  I do find it much easier to maintain one file for a cms platform for a deployer recipe. 
- The idea of separate classes in separate files is good if you are developing a platform with many classes.  A C++ program, or a PHP platform system in which your editor can quickly assist you in finding class definitions and references via a ctags file is VERY nice. However the scope of deployer-recipes is just not that big in my opinion to warrant such file&class separation.  Furthermore, when I had the files separated out ([see earlier commits](https://github.com/unleashedtech/deployer-recipes/pull/14/commits/1a4de5d24d3e57b8a55fe3d2551f446c304aac36)) having such small class and file definitions (20 files each with only 15 lines in them) was very cumbersome and not useful.  

- I believe the initial intent was to use Deployer7's ability to inherit cascading yml files and the end result being actually less lines of code.  And that is also a good thing.  However in practice, because of some of the limitations of yml, ALL my classes ended up being in php and so I saw no need to keep the yml files around or the small task oriented files.  
- Another note is that when I had multiple files for tasks and I would want to re-arrange the hierarchy of a task (to take a db related task out of magento specific related directory structure for example) it was very cumbersome to have to rename directories and move many files.  In contrast to that, when a task re-arrangement was necessary and I had everything in one file, it took seconds complete the re-arrangement.
- Finally, I looked to the community who had already completed a magento2.php file ( https://github.com/deployphp/deployer/blob/master/recipe/magento2.php ) and really liked the one file method because it was much more straight forward, easier to debug, easier to read .... And the deployerphp platform is our target audience..

Reply:
Noted! Yes, we'll have to continue weighing whether the unified approach used for this platform should be the standard for all supported platforms in this package. We can revisit this discussion later.